Agents/Xamarin.Interactive/CodeAnalysis/Evaluating/EvaluationContextManager.cs

```diff
@@ -65,8 +65,9 @@ public void Dispose ()
         readonly Observable<ICodeCellEvent> events = new Observable<ICodeCellEvent> ();
         public IObservable<ICodeCellEvent> Events => events;
 
-        public IAgentSynchronizationContext SynchronizationContexts { get; }
-            = new AgentSynchronizationContext ();
+        readonly Lazy<AgentSynchronizationContext> agentSynchronizationContext
+            = new Lazy<AgentSynchronizationContext> ();
+        public virtual IAgentSynchronizationContext SynchronizationContexts => agentSynchronizationContext.Value;
```

Comment:
This lets the WASM ECM control this during actual evaluations—there's no sync context in WASM, so trying to use the default value causes it to throw up. In brief testing, using the single-thread sync context that we use on the Console app causes lockups, so I preferred to just leave this null (see `WebAssemblyEvaluationContextManager` in the workbook app).